common/src/main/java/com/wynntils/features/user/tooltips/ItemCompareFeature.java

```diff
@@ -0,0 +1,154 @@
+        if (!(McUtils.mc().screen instanceof AbstractContainerScreen<?> abstractContainerScreen)) return;
+
+        if (abstractContainerScreen.hoveredSlot == null) {
+            return;
+        }
+
+        ItemStack hoveredItemStack = abstractContainerScreen.hoveredSlot.getItem();
+
+        if (event.getItemStack() != hoveredItemStack) {
+            return;
+        }
+
+        if (!(hoveredItemStack instanceof GearItemStack hoveredGearItemStack)) {
+            return;
+        }
+
+        GearItemStack itemToCompare = null;
+
+        // No compared item selected, try compare to equipped armor
+        if (compareToEquipped) {
+            List<ItemStack> armorSlots = McUtils.player().getInventory().armor;
+
+            Optional<GearItemStack> matchingArmorItemStack = armorSlots.stream()
+                    .filter(itemStack -> itemStack instanceof GearItemStack gItemStack
+                            && gItemStack.getItemProfile().getItemInfo().getType()
+                                    == hoveredGearItemStack
+                                            .getItemProfile()
+                                            .getItemInfo()
+                                            .getType())
+                    .map(itemStack -> (GearItemStack) itemStack)
+                    .findFirst();
+
+            itemToCompare = matchingArmorItemStack.orElse(null);
+        } else if (comparedItem != null) {
+            itemToCompare = comparedItem;
+        }
+
+        if (itemToCompare == null) {
+            return;
+        }
+
+        if (itemToCompare == hoveredGearItemStack) {
+            itemToCompare = null;
+        }
+
+        if (itemToCompare == null) return;
+
+        final PoseStack poseStack = new PoseStack();
+        final int mouseX = event.getMouseX();
+        final int mouseY = event.getMouseY();
+
+        poseStack.pushPose();
+
+        poseStack.translate(0, 0, 300);
+
+        int toBeRenderedWidth = abstractContainerScreen.getTooltipFromItem(itemToCompare).stream()
+                .map(component -> McUtils.mc().font.width(component))
+                .max(Integer::compareTo)
+                .orElse(0);
+
+        int hoveredWidth = abstractContainerScreen.getTooltipFromItem(hoveredGearItemStack).stream()
+                .map(component -> McUtils.mc().font.width(component))
+                .max(Integer::compareTo)
+                .orElse(0);
+
+        if (mouseX + toBeRenderedWidth + hoveredWidth > abstractContainerScreen.width) {
+            abstractContainerScreen.renderTooltip(poseStack, itemToCompare, mouseX - toBeRenderedWidth - 10, mouseY);
+        } else {
+            abstractContainerScreen.renderTooltip(poseStack, itemToCompare, mouseX + hoveredWidth + 10, mouseY);
+        }
+
+        poseStack.popPose();
+    }
+
+    @SubscribeEvent
+    public void onScreenClose(ScreenClosedEvent event) {
+        compareToEquipped = false;
```

Comment:
Qodana is still b*tching about that the variables are static. I don't think there is any reason to, our features are created as singletons.